modules/apps/commerce/commerce-product-type-virtual-service/src/main/java/com/liferay/commerce/product/type/virtual/service/impl/CPDVirtualSettingFileEntryLocalServiceImpl.java

```diff
@@ -194,39 +191,9 @@ public CPDVirtualSettingFileEntry updateCPDVirtualSettingFileEntry(
 			cpdVirtualSettingFileEntryPersistence.update(
 				cpdVirtualSettingFileEntry);
 
-		if (fileEntryId != oldCPDVirtualSettingFileEntryFileEntryId) {
-			_deleteFileEntry(oldCPDVirtualSettingFileEntryFileEntryId);
-		}
-
 		return cpdVirtualSettingFileEntry;
 	}
 
-	private void _deleteFileEntry(
-		CPDVirtualSettingFileEntry cpdVirtualSettingFileEntry) {
-
-		_deleteFileEntry(cpdVirtualSettingFileEntry.getFileEntryId());
-	}
-
-	private void _deleteFileEntry(long cpdVirtualSettingFileEntryFileEntryId) {
-		try {
-			if (cpdVirtualSettingFileEntryFileEntryId <= 0) {
-				return;
-			}
-
-			int countCPDVirtualSettingFileEntryByFileEntryId =
-				cpdVirtualSettingFileEntryLocalService.countByFileEntryId(
-					cpdVirtualSettingFileEntryFileEntryId);
-
-			if (countCPDVirtualSettingFileEntryByFileEntryId == 0) {
-				_dlAppLocalService.deleteFileEntry(
-					cpdVirtualSettingFileEntryFileEntryId);
-			}
```

Comment:
in reality here we were deleting the file only if not attached to any cpdVirtualSettingFileEntry. Why we cannot extend this logic to say delete if not attached to any cpdVirtualSettingFileEntry or orderItem?